src/fidasim.f90

```diff
@@ -2726,9 +2804,10 @@ subroutine read_equilibrium
     !!Close HDF5 interface
     call h5close_f(error)
 
-    allocate(equil%mask(inter_grid%nr,inter_grid%nz))
+    allocate(equil%mask(inter_grid%nr,inter_grid%nz,inter_grid%nphi))
     equil%mask = 0.d0
     where ((p_mask.eq.1).and.(f_mask.eq.1)) equil%mask = 1.d0
+    where (p_mask.eq.1) equil%mask = 1.d0
```

Comment:
Remove this line. Both the plasma parameters and the fields need to be defined, which is what the preceding line does.